sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBSuite.scala

```diff
@@ -2240,6 +2240,166 @@ class RocksDBSuite extends AlsoTestWithChangelogCheckpointingEnabled with Shared
     }
   }
 
+  test("Rocks DB task completion listener does not double unlock acquireThread") {
+    // Create a custom ExecutionContext with 3 threads
```

Comment:
nit: is num of threads 3 or 5?

Reply:
hmm, still showing 5 for me: `ThreadUtils.newDaemonFixedThreadPool(5, "pool-thread-executor"))`